data/transaction/status_test.go

```diff
@@ -11,112 +11,121 @@ import (
+	tx.DestinationShard = 13
+	responseStatus, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.TxBlock, tx)
+	require.Equal(t, TxStatusPending, responseStatus)
+	require.Nil(t, err)
 
 	// Cross, at source, but knowing that it has been fully notarized (through DatabaseLookupExtensions)
-	computer = &StatusComputer{
-		MiniblockType:        block.TxBlock,
-		IsMiniblockFinalized: true,
-		SourceShard:          12,
-		DestinationShard:     13,
-		SelfShard:            12,
-	}
-	require.Equal(t, TxStatusSuccess, computer.ComputeStatusWhenInStorageKnowingMiniblock())
+	tx.DestinationShard = 13
+	tx.SourceShard = 12
+	tx.NotarizedAtDestinationInMetaNonce = 1
+	responseStatus, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.TxBlock, tx)
+	require.Equal(t, TxStatusSuccess, responseStatus)
+	require.Nil(t, err)
 
 	// Cross, destination me
-	computer = &StatusComputer{
-		MiniblockType:    block.TxBlock,
-		SourceShard:      13,
-		DestinationShard: 12,
-		SelfShard:        12,
-	}
-	require.Equal(t, TxStatusSuccess, computer.ComputeStatusWhenInStorageKnowingMiniblock())
-
-	computer = &StatusComputer{
-		MiniblockType:    block.RewardsBlock,
-		SourceShard:      core.MetachainShardId,
-		DestinationShard: 12,
-		SelfShard:        12,
-	}
-	require.Equal(t, TxStatusSuccess, computer.ComputeStatusWhenInStorageKnowingMiniblock())
+	tx.DestinationShard = 12
+	tx.SourceShard = 13
+	tx.NotarizedAtDestinationInMetaNonce = 0
+	responseStatus, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.TxBlock, tx)
+	require.Equal(t, TxStatusSuccess, responseStatus)
+	require.Nil(t, err)
+
+	tx.SourceShard = core.MetachainShardId
+	tx.DestinationShard = 12
+	tx.NotarizedAtDestinationInMetaNonce = 0
+	responseStatus, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.RewardsBlock, tx)
+	require.Equal(t, TxStatusSuccess, responseStatus)
+	require.Nil(t, err)
 
 	// Contract deploy
-	computer = &StatusComputer{
-		SourceShard:      12,
-		DestinationShard: 13,
-		Receiver:         make([]byte, 32),
-		TransactionData:  []byte("deployingAContract"),
-		SelfShard:        13,
-	}
-	require.Equal(t, TxStatusSuccess, computer.ComputeStatusWhenInStorageKnowingMiniblock())
+	statusComputer.SelfShardId = 13
+	tx.SourceShard = 12
+	tx.DestinationShard = 13
+	tx.Tx.SetRcvAddr(make([]byte, 32))
+	tx.Data = []byte("deployingAContract")
+	responseStatus, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.TxBlock, tx)
+	require.Equal(t, TxStatusSuccess, responseStatus)
+	require.Nil(t, err)
+
+	// Nil parameters
+	_, err = statusComputer.ComputeStatusWhenInStorageKnowingMiniblock(block.TxBlock, nil)
+	require.Equal(t, ErrNilApiTransactionResult, err)
 }
 
 func TestStatusComputer_ComputeStatusWhenInStorageNotKnowingMiniblock(t *testing.T) {
-	// Intra shard
-	computer := &StatusComputer{
+	statusComputer := &StatusComputer{
+		SelfShardId: 12,
+	}
+
```

Comment:
same, why don't you use the constructor ?

Reply:
Done!